doc: Azure doc fixes

umapraseeda opened the pull request:
Azure doc fixes

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

umapraseeda committed:
doc: Review of Azure AD B2C document

Review of Azure AD B2C document

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+   #. Select :guilabel:`Platform Settings` > :guilabel:`Authentication/Authorization` > :guilabel:`Azure Active Directory Settings`.
+
+#. Enter the B2C Application ID into the :guilabel:`Client ID` field.
+#. Enter the B2C Open ID Connect metadata URL into the :guilabel:`Issuer URI` field and click ``OK`` to finish the configuration.
```
```suggestion
#. Enter the B2C OpenID Connect metadata URL into the :guilabel:`Issuer URI` field and click :guilabel:`OK` to finish the configuration.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
 
-#.  Validation and Testing.
-    The configuration on both the B2C application side as well as the Azure Function is now complete.
+   #. Select the entry that corresponds to your Azure function app in the :guilabel:`Application` field.
```
```suggestion
   #. Select the entry that corresponds to your Azure Function App in the :guilabel:`Application` field.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+   a. Clear the browser session cookies and confirm that the user needs to authenticate during an attempt to access your Azure function.
+   #. After a successful access to the function app, open a separate browser tab and validate that you can automatically sign in.
+   #. Run other user flows, such as password reset or profile edit, with your Azure function.
+   #. After the users complete these flows, they are redirected to the Azure function app.
```
```suggestion
   #. After the users complete these flows, they are redirected to the Azure Function App.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-If you want to use Azure Function Apps and need a way to secure access to your Function Apps -- that is, only allowing authenticated users, then Azure AD B2C is a great solution.
-It allows users to create their own accounts or using a social identity provider, and gain access to the your Function apps only after they have successfully authenticated.
-This document is a step-by-step guide to configuring Azure AD B2C, as the identity provider for your Azure Function Apps.
+To secure access to your Azure function apps and to allow access only to authenticated users, you can use Azure Active Directory B2C.
```
```suggestion
To secure access to your Azure Function App and to allow access only to authenticated users, you can use Azure Active Directory B2C.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-    .. image:: ./secure-azure-function-apps-with-microsoft-b2c/01.png
-       :alt: 1
+#. Log in to the `Azure portal <https://portal.azure.com/#home>`_.
+#. Click :guilabel:`+Create a Resource` to create your Azure function app.
```
```suggestion
#. Click :guilabel:`+Create a Resource` to create your Azure Function App.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+	
+	  * Provide the app name.
+	  * Select ``Yes`` for the :guilabel:`Web App/ Web API` option.
+	  * In the :guilabel:`Reply URL` field, enter the unique Azure function app name that you created in step 4 and append  ``/.auth/login/aad/callback`` to it. For example, ``https://myAppSecuredByB2C.azurewebites.net/.auth/login/aad/callback``.
```
```suggestion
	  * In the :guilabel:`Reply URL` field, enter the unique Azure Function App name that you created in step 4 and append  ``/.auth/login/aad/callback`` to it. For example, ``https://myAppSecuredByB2C.azurewebites.net/.auth/login/aad/callback``.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+      :alt: 1
+
+
+#. Enter a unique name for your function app, fill the remaining information and click :guilabel:`Create`.
```
```suggestion
#. Enter a unique name for your Function App, fill the remaining information and click :guilabel:`Create`.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
 
-#.  Go back to the Function App blade in the Azure Portal, and select the Azure function app that you created.
-    From the Function App blade, select Platform Features and then select Authentication/Authorization.
+   Once the function app is created, click :guilabel:`Go to resource` to view the created function app.
```
```suggestion
   Once the Function App is created, click :guilabel:`Go to resource` to view the created Function App.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+    
+   .. note:: 
+	
+	  The default action when a request is not authenticated is ``Allow Anonymous request``. This allows any user to access your function app.
```
```suggestion
	  The default action when a request is not authenticated is ``Allow Anonymous request``. This allows any user to access your Function App.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+
+#. Navigate to :guilabel:`All resources` in the Azure portal home page.
+
+   a. Click the function app that you created.
```
```suggestion
   a. Click the Function App that you created.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-#.  Return to the Azure Function blade, and navigate to the Platform Settings -> Authentication/Authorization -> Azure Active Directory Settings -- enter the B2C Application ID into the **Client ID** property.
-    Similarly, enter in the B2C Open ID Connect metadata URL into the **Issuer URI** property.
-    Select OK to finish the configuration.
+1. Confirm that users without authentication gets prompted to complete their B2C sign-in before being allowed access to your function app:
```
```suggestion
1. Confirm that users without authentication gets prompted to complete their B2C sign-in before being allowed access to your Function App:
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-        #.  After successful Function App access, try opening a separate browser tab, and validate that you can automatically sign-in.
-        #.  Try to run other user flows, such as password reset or profile edit, with your Azure Function -- after the users complete these flows, the will be redirected to the Azure Function App.
+   a. Clear the browser session cookies and confirm that the user needs to authenticate during an attempt to access your Azure function.
+   #. After a successful access to the function app, open a separate browser tab and validate that you can automatically sign in.
```
```suggestion
   #. After a successful access to the Function App, open a separate browser tab and validate that you can automatically sign in.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+      :alt: 4
 
-#.  We will now select "Log in with Azure Active Directory", which is one of the available options in the drop down box labeled "Action to take when request is not authenticated"
+#. Select ``Log in with Azure Active Directory`` in the :guilabel:`Action to take when request is not authenticated` field.
```
```suggestion
#. Select :guilabel:`Log in with Azure Active Directory` in the :guilabel:`Action to take when request is not authenticated` field.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-#.  Enter your unique Azure Function App name -- copy the name to a text editor -we will need it in subsequent steps.
-    Complete the rest of the information and select **Create**.
+   a. Search for ``function app``.
+   #. In the ``Function App`` blade, click :guilabel:`Create`. 
```
```suggestion
   #. In the :guilabel:`Function App` blade, click :guilabel:`Create`. 
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
 
-#.  Move the App Service Authentication to the On Position, which then expose a number of settings available for Authentication.
-    Note that the default is "Allow Anonymous request" -- allowing any user to access your function application.
+#. In the :guilabel:`Authentication/Authorization` blade, move the sliding button under :guilabel:`App Service Authentication` to ``On``. Several authentication provider configurations are now displayed.
```
```suggestion
#. In the :guilabel:`Authentication/Authorization` blade, move the sliding button under :guilabel:`App Service Authentication` to :guilabel:`On`. Several authentication provider configurations are now displayed.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
 
-#.  After this select "Azure Active Directory" under the list of Authentication Providers, this will open a new configuration blade, select "Advanced" and note the required fields, including Client ID and Issuer Uri -- the Client Secret and Allowed Token Audiences are option fields.
-    Go to the next steps to collect the application Client ID and Issuer URI values from B2C.
+#. Select ``Azure Active Directory`` under the :guilabel:`Authentication Providers` list.
```
```suggestion
#. Select :guilabel:`Azure Active Directory` under the :guilabel:`Authentication Providers` list.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+   #. Configure the following fields:
+	
+	  * Provide the app name.
+	  * Select ``Yes`` for the :guilabel:`Web App/ Web API` option.
```
```suggestion
	  * Select :guilabel:`Yes` for the :guilabel:`Web App/ Web API` option.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-######################################################
+.. _azure-adb2c:
 
-Manual Guide by `Saurabh Sharma <https://docs.microsoft.com/answers/answers/40449/view.html>`_.
```
Original author removed.

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+   #. Click on :guilabel:`Applications` and then click on the entry for your newly created application (``B2CFunctionApp`` in this example) to open the application profile.
+   #. Obtain the Application ID from the Properties blade of the application.
+
+   Use this ID to complete the Azure function configuration.
```
```suggestion
   Use this ID to complete the Azure Function configuration.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-        #.  Clear browser session cookies, validate that user must authenticate when attempt to access your Azure Function.
-        #.  After successful Function App access, try opening a separate browser tab, and validate that you can automatically sign-in.
-        #.  Try to run other user flows, such as password reset or profile edit, with your Azure Function -- after the users complete these flows, the will be redirected to the Azure Function App.
+   a. Clear the browser session cookies and confirm that the user needs to authenticate during an attempt to access your Azure function.
```
```suggestion
   a. Clear the browser session cookies and confirm that the user needs to authenticate during an attempt to access your Azure Function.
```

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-        #.  Try to run other user flows, such as password reset or profile edit, with your Azure Function -- after the users complete these flows, the will be redirected to the Azure Function App.
+   a. Clear the browser session cookies and confirm that the user needs to authenticate during an attempt to access your Azure function.
+   #. After a successful access to the function app, open a separate browser tab and validate that you can automatically sign in.
+   #. Run other user flows, such as password reset or profile edit, with your Azure function.
```
```suggestion
   #. Run other user flows, such as password reset or profile edit, with your Azure Function.
```

coderbyheart reviewed (changes requested):
- `Azure Function App`/`Azure Function` is a product name, capitalize in all instances
- use guilabel for all GUI elements

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
 
-    .. image:: ./secure-azure-function-apps-with-microsoft-b2c/02.png
+   .. image:: ./secure-azure-function-apps-with-microsoft-b2c/01.png
+      :alt: 1
```
Do we want to have a caption for the image?

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-       :alt: 3
+   .. image:: ./secure-azure-function-apps-with-microsoft-b2c/03.png
+      :alt: 3
 
```
Caption?

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+
+   .. image:: ./secure-azure-function-apps-with-microsoft-b2c/06.png
        :alt: 6
 
```
Caption?

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+
+   .. image:: ./secure-azure-function-apps-with-microsoft-b2c/07.png
+      :alt: 7
+
```
Caption?

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+
+   a. In the Azure portal, select the :guilabel:`Azure AD B2C` blade.
+   #. Click on :guilabel:`Applications` and then click on the entry for your newly created application (``B2CFunctionApp`` in this example) to open the application profile.
+   #. Obtain the Application ID from the Properties blade of the application.
```
```suggestion
   #. Obtain the Application ID from the :guilabel:`Properties` blade of the application.
```

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+
+   .. image:: ./secure-azure-function-apps-with-microsoft-b2c/08.png
+      :alt: 8
+
```
Caption?

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+#. Create a new B2C application:
+
+   a. In the Azure Portal (open a separate browser tab or new browser session if you want the previous blade to be available), select the :guilabel:`Azure AD B2C` blade.
+   #. Select :guilabel:`Applications`, then select :guilabel:`+Add`.
```
```suggestion
   #. Select :guilabel:`Applications` > :guilabel:`+Add`.
```

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-    .. image:: ./secure-azure-function-apps-with-microsoft-b2c/08.png
-       :alt: 8
+   .. image:: ./secure-azure-function-apps-with-microsoft-b2c/10.png
+      :alt: 10
```
Caption?

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-        Make sure to select the Application representing your Azure Function App in the Application drop down picker dialog, and the associated Reply URL (in some cases you many have more than one reply URL).
+      .. image:: ./secure-azure-function-apps-with-microsoft-b2c/11.png
+         :alt: 11
 
```
Caption?

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
+Users can access the function apps only after a successful authentication.
 
-#.  Create your Azure function app -- easiest way is to select **+Create a Resource** , type in "function app" to find it, select Function App, and select **Create**
+To configure Azure AD B2C as the identity provider for your function apps, complete the following steps:
```
Another instance of "function apps".

peknis commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,138 @@
-This document is a step-by-step guide to configuring Azure AD B2C, as the identity provider for your Azure Function Apps.
+To secure access to your Azure function apps and to allow access only to authenticated users, you can use Azure Active Directory B2C.
+It allows users to create their own accounts or sign in with credentials provided by a social identity provider.
+Users can access the function apps only after a successful authentication.
```
Do we need to capitalize also this instance of "function apps"?

wiba-nordic committed:
doc: review of Azure AD B2C

Revisions based on following the instructions on the current Azure portal as far as I could.
I also removed the pictures since they were outdated with the current design of the portal.

Signed-off-by: Wille Backman <wille.backman@nordicsemi.no>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/NordicSemiconductor/asset-tracker-cloud-docs?pullRequest=174) <br/>All committers have signed the CLA.

wiba-nordic commented:
I took out the illustrations, because when I was following the instructions, they weren't really helping and they were out of date with the updated Azure UI.

peknis commented:
The doc has now been stripped off all images. A few images might help the reader to navigate, and at least gives a bit of breath in a long procedure like this. Also, check again that all instances of Function App are capitalized.

peknis commented:
> * `Azure Function App`/`Azure Function` is a product name, capitalize in all instances
> * use guilabel for all GUI elements

However, one could argue whether the Function App needs to be capitalized or not. See for example MS's own docs: https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-function-app-portal

wiba-nordic committed:
Restructure and small fixes

Restructured some of the instructions to remove duplicated steps.
Added some updated images to break the wall of text a bit.
Also capitalized all instances of Function App.

Signed-off-by: Wille Backman <wille.backman@nordicsemi.no>

wiba-nordic committed:
final polishing

Added some more polish, and fixed a wrong image link.

Signed-off-by: Wille Backman <wille.backman@nordicsemi.no>

coderbyheart commented on docs/azure/secure-azure-function-apps-with-microsoft-b2c.rst:
```diff
@@ -1,86 +1,124 @@
-#.  Enter your unique Azure Function App name -- copy the name to a text editor -we will need it in subsequent steps.
-    Complete the rest of the information and select **Create**.
+   a. Search for ``function app``.
+   #. Under ``Function App``, click :guilabel:`Create`.
```
```suggestion
   #. Under :guilabel:`Function App`, click :guilabel:`Create`.
```
